pkg/helm/watcher/controller/helm_watcher.go

```diff
@@ -138,3 +158,84 @@ func (r *HelmWatcherReconciler) reconcileDelete(ctx context.Context, repository
 	// Stop reconciliation as the object is being deleted
 	return ctrl.Result{}, nil
 }
+
+// sendEvent emits a Kubernetes event and forwards the event to notification controller if configured.
+func (r *HelmWatcherReconciler) sendEvent(ctx context.Context, hr *sourcev1.HelmRepository, revision, severity, msg string) {
+	log := logr.FromContextOrDiscard(ctx)
+
+	if r.ExternalEventRecorder == nil {
+		return
+	}
```

Comment:
Is this because notifications is optional, or because this could be unset? If its the latter this is smelly... can we make `HelmWatcherReconciler` fields private, have a constructor instead and then do any validation of fields (or defaulting) done in there?

Reply:
Optional. If not set, we don't send an event. It's only set if the address is set. If set specifically to empty, then we don't send any events.

Reply:
isn't the address defaulting to a value, so its always set?

Reply:
Yes, hence, if you don't want it you have to explicitly set it to empty.